src/main/java/org/opensearch/knn/index/codec/util/VectorTransferByte.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.util;
+
+import org.opensearch.knn.jni.JNICommons;
+
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class VectorTransferByte extends VectorTransfer {
+    private List<byte[]> vectorList;
+
+    public VectorTransferByte(final long vectorsStreamingMemoryLimit) {
+        super(vectorsStreamingMemoryLimit);
+        vectorList = new ArrayList<>();
+    }
+
+    @Override
+    public void init(final long totalLiveDocs) {
+        this.totalLiveDocs = totalLiveDocs;
+        vectorList.clear();
+    }
+
+    @Override
+    public void addVector(final ByteArrayInputStream byteStream) {
```

Comment:
the function is doing more than 1 thing, it is adding vectors and also streaming the vectors. We should break this either into 2 functions or rename the function in a fashion that it provides this information that we are doing 2 operations.